scarlet/observation.py

```diff
@@ -259,133 +272,105 @@ def __init__(self, images, wcs=None, psfs=None, weights=None, filtercurve=None,
-
-                _target = scene.psfs[0, :, :]
-                _shape = scene.shape
-
-            else:
-                _target = self.target_psf
-
-            resconv_op = []
-
-            for _psf in self.psfs:
-                # Computes spatially matching observation and target psfs. The observation psf is also resampled to the scene's resolution
-
-                new_target, observed_psf = resampling.match_psfs(_target, _psf, whr, wlr)
-                # Computes the diff kernel in Fourier
-                target_fft = np.fft.fft2(np.fft.ifftshift(new_target))
-                observed_fft = np.fft.fft2(np.fft.ifftshift(observed_psf))
-                kernel_fft = np.zeros(target_fft.shape)
-                sel = target_fft != 0
-                kernel_fft[sel] = observed_fft[sel] / target_fft[sel]
-                kernel = np.fft.ifft2(kernel_fft)
-                kernel = np.fft.fftshift(np.real(kernel))
-                diff_psf = kernel / kernel.max()
-
-                # Computes the resampling/convolution matrix
-                resconv_op.append(resampling.make_operator(_shape, coord_hr, diff_psf))
-
-            self.resconv_op = np.array(resconv_op)
-
-        else:
-            class InitError(Exception):
-                '''
-                'Observation PSF needed: unless you are not dealing with astronomical data, you are doing something wrong'
-                '''
-                pass
-
-            raise InitError
+    def match(self, model_frame):
+
+        #  bands of model that are represented in this observation
+        self.band_slice = slice(None)
+        if self.frame.bands is not model_frame.bands:
+            bmin = model_frame.bands.index(self.frame.bands[0])
+            bmax = model_frame.bands.index(self.frame.bands[-1])
+            self.band_slice = slice(bmin, bmax+1)
+
+        # Get pixel coordinates in each frame.
+        mask, coord_lr, coord_hr = resampling.match_patches(model_frame.shape, self.frame.shape, model_frame.wcs, self.frame.wcs)
+        self._coord_lr = coord_lr
+        self._coord_hr = coord_hr
+        self._mask = mask
+
+        # Compute diff kernel at hr
+
+        whr = model_frame.wcs
+        wlr = self.frame.wcs
+
+        # Reference PSF
+        _target = model_frame.psfs[0, :, :]
+        _shape = model_frame.shape
+
+        resconv_op = []
+        for _psf in self.frame.psfs:
+            # Computes spatially matching observation and target psfs. The observation psf is also resampled to the model frame resolution
+            new_target, observed_psf = resampling.match_psfs(_target, _psf, whr, wlr)
+            # Computes the diff kernel in Fourier
+            target_fft = np.fft.fft2(np.fft.ifftshift(new_target))
+            observed_fft = np.fft.fft2(np.fft.ifftshift(observed_psf))
+            kernel_fft = np.zeros(target_fft.shape)
+            sel = target_fft != 0
+            kernel_fft[sel] = observed_fft[sel] / target_fft[sel]
```

Comment:
This line generates a `ComplexWarning`:
```
/Users/fred/projects/scarlet/scarlet/observation.py:308: ComplexWarning: Casting complex values to real discards the imaginary part
  kernel_fft[sel] = observed_fft[sel] / target_fft[sel]
```
I think this is because `kernel_fft` is not initialized as a complex array, which it needs to be because `observed_fft` and `target_fft` are both complex. This might solve the color issue that you described in slack.

Reply:
Problem for @herjy to fix in the new FFT code

Reply:
It is already fixed on my branch